dm/master/server.go

```diff
@@ -1334,37 +1341,232 @@ func (s *Server) generateSubTask(ctx context.Context, task string) (*config.Task
 }
 
 var (
-	maxRetryNum = 30
+	maxRetryNum   = 30
+	retryInterval = time.Second
 )
 
-// taskConfigArgsExtractor extracts SubTaskConfig from args and returns its relevant
-// grpc client, worker id (host:port), subtask config in toml, task name and error
-func (s *Server) taskConfigArgsExtractor(cfg *config.SubTaskConfig) (*scheduler.Worker, string, string, error) {
-	handleErr := func(err error) error {
-		log.L().Error("response", zap.Error(err))
-		return err
+func extractWorkerError(result *pb.ProcessResult) error {
+	if result != nil && len(result.Errors) > 0 {
+		return terror.ErrMasterOperRespNotSuccess.Generate(result.Errors[0].String())
```

Comment:
how about adding all error's information in `result.Errors`?

Reply:
revised in https://github.com/pingcap/dm/pull/490/commits/6d43a1ddce1f583c1a734b38e787d65cc9ad6261